File: src/main/java/arimitsu/sf/cql/v3/Flags.java
package arimitsu.sf.cql.v3;

/**
 * Created by sxend on 14/06/04.
 */
public enum Flags {

    NONE(0x00),
    COMPRESSION(0x01),
    TRACING(0x02),
    BOTH(0x03),;

    public final byte value;

    Flags(int value) {
        this.value = (byte) value;
    }

    public static Flags valueOf(byte value) {
        for (Flags v : values()) {
            if (v.value == value) return v;
        }
        throw new RuntimeException("invalid value");
    }
}

File: src/main/scala/arimitsu/sf/cql/v3/Flags.scala
package arimitsu.sf.cql.v3

;

/**
 * Created by sxend on 14/06/04.
 */
object Flags {

  sealed abstract class Flags(val value: Byte)

  case object NONE extends Flags(0x00)

  case object COMPRESSION extends Flags(0x01)

  case object TRACING extends Flags(0x02)

  case object BOTH extends Flags((COMPRESSION.value | TRACING.value).toByte)

}
